Yoast/wordpress-seo
# Indexable system page

## Summary

Refactors the search result builder and watcher into system page builder and watcher. This allows them to also handle 404 pages and, possibly in the future, things like login pages, cart pages etc.

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Made builders more consistent by having them always set the `object_id` and `object_type` themselves.

## Test instructions
This PR can be tested by following these steps:

* Visit a search result page. You should see a title in the new frontend output if you've set one in search appearance.
* Visit a 404 page. You should see a title in the new frontend output if you've set one in search appearance.
* Change your settings in search appearance.
* Your change should be reflected on both pages.

## UI changes

## Documentation

## Quality assurance
